python/whylogs/core/resolvers.py

```diff
@@ -11,11 +12,45 @@
 ColumnSchema: TypeAlias = "ColumnSchema"  # type: ignore
 
 
+AdditionalMetric = namedtuple("AdditionalMetric", "metric column_names why_types")
+
+
 class Resolver(ABC):
     """A resolver maps from a column name and a data type to trackers.
 
     Note that the key of the result dictionaries defines the namespaces of the metrics in the serialized form."""
 
+    def __init__(self):
+        self.additional_metrics: List[AdditionalMetric] = []
+
+    def add_standard_metric(
+        self,
+        metric: StandardMetric,
+        column_names: List[String] = [],
+        why_types: List[Union[Fractional, Integral, String]] = [],
+    ):
```

Comment:
Ok, so this can only add a new `StandardMetric`, that wouldn't already resolve for this type or column name on a resolver, so its rather limited functionally on the `StandardResolver `since you would only be able to add a `StandardMetric`, I guess the main use case here is the `unicode_range `and `condition_count `metrics?

Also its a bit confusing what happens when I add the same `StandardMetric` that would already be resolved, or the same `StandardMetric` type gets added multiple times, perhaps bound to a type and then bound to a column name? It looks like they can both be added but only one will be functional, so maybe it doesn't break things but slows down resolving column schema?

Reply:
With latest changes, this now expects a SpecResolver, which accepts any kind of Metrics, not just Standard ones.

If the metric to be added is already present in the column's result while resolving the additional specs, it will just skip it.

If both a name and type is passed, the name takes precedence, and will only be added once